src/main/java/com/thoughtworks/pacman/ui/screens/IntroScreen.java

```diff
@@ -1,28 +1,114 @@
 package com.thoughtworks.pacman.ui.screens;
 
-import java.awt.Dimension;
+import java.awt.*;
 import java.awt.Graphics2D;
 import java.awt.Image;
-import java.awt.event.KeyEvent;
+import java.awt.event.*;
+import java.awt.Color;
+import java.awt.Rectangle;
+
+import java.awt.font.TextLayout;
+
 
 import com.thoughtworks.pacman.core.Game;
 import com.thoughtworks.pacman.ui.ImageLoader;
 import com.thoughtworks.pacman.ui.Screen;
 
+
 public class IntroScreen implements Screen {
     static final Image TITLE_SCREEN_IMAGE = ImageLoader.loadImage(Screen.class, "titleScreen.jpg");
 
     private final Dimension dimension;
     private boolean startGame;
+    private State currentStateofPlayButton = State.RELEASED_PLAY_BUTTON;
+    private State currentStateOfSettingsButton = State.RELEASED_SETTINGS_BUTTON;
+    private State currentStateOfQuitButton = State.RELEASED_QUIT_BUTTON;
+    private Color buttonMainColor; // main color theme: Yellow
+    private Color buttonOnHoverColor; // when on hover color theme:Darker Yellow
+    
+    private Rectangle playClickBox;
+    private Rectangle settingsClickBox;
+    private Rectangle exitClickBox;
+    
 
     public IntroScreen(Game game) {
         this.dimension = game.getDimension();
+        buttonMainColor = new Color(255, 255, 0);
+        buttonOnHoverColor = new Color(156, 156, 2);
+        playClickBox = new Rectangle(180, 310, 100, 30);
+        settingsClickBox = new Rectangle(180, 350, 100, 30);
+        exitClickBox =new Rectangle(180, 390, 100, 30);
         this.startGame = false;
+
     }
 
     public void draw(Graphics2D graphics) {
         int height = TITLE_SCREEN_IMAGE.getHeight(null) * dimension.width / TITLE_SCREEN_IMAGE.getWidth(null);
         graphics.drawImage(TITLE_SCREEN_IMAGE, 0, 0, dimension.width, height, null);
+
+        if(currentStateofPlayButton == State.RELEASED_PLAY_BUTTON){
+            graphics.setColor(buttonMainColor);
```

Comment:
Refactor these if blocks. See if you reuse any code and move that into a method/methods with meaningful names. Then call those methods.